HangMan/src/hangman.rs

```diff
@@ -0,0 +1,139 @@
+use std::vec;
+use dioxus::prelude::*;
+
+//component for managing the image of the hangman
+#[inline_props]
+pub fn Hangman(cx:Scope, stage: usize, reset_image: bool) -> Element {
+    
+    let css_status = use_ref(cx, || {vec![true;11]});    
+
+    //changes the hidden status of the next image in queque when the user makes a guess
+    use_memo(cx, stage, |stage| {
+        
+        css_status.with_mut(|current_list| {
+            if stage >= current_list.len() {
+                panic!("SHOULDNT ALLOW THE USER TO MAKE MORE THAN 10 GUESSES");
+            } else {
+                current_list[stage] = false;
+            }
+        })   
+    });
+
+    use_memo(cx, reset_image, |_| {
+        css_status.with_mut(|cur| {
+            *cur = vec![true;11];
+        })
+    });
+
+    render!(rsx!(
+        link { rel: "stylesheet", href: "HangMan/input.css"}
+        div {
+            class: "hangman-container",
+
+            css_status.read().iter().skip(1).enumerate().map(move |(idx,state)| {
+                rsx!(
+                    div {
+                        class: "hangman{(idx)}",
+                        hidden: *state,
+                    }
+                )
+            })            
+        }
+
+    ))
+}
+#[allow(non_upper_case_globals)]
+#[inline_props]
+pub fn KeyBoard<'a>(cx: Scope, onguess: EventHandler<'a, char>, reset_game: bool, disable_all: bool) -> Element<'a> {
+    const row_1_letters: [char;10] = ['q','w','e','r','t','y','u','i','o','p'];
```

Comment:
There is a lot of duplication here with three differently sized rows.

You can combine all of these arrays like this
```suggestion
    const row_letters: &[&[char]] = &[
        &['q','w','e','r','t','y','u','i','o','p'],
        &['a','s','d','f','g','h','j','k','l'],
        &['z','x','c','v','b','n','m']
    ];
```